Storage/SQL/BrowserDB.swift

```diff
@@ -32,6 +32,16 @@ open class BrowserDB {
         self.db = SwiftData(filename: file, key: secretKey, prevKey: nil, schema: schema, files: files)
     }
 
+    // Remove the DB op from the queue (by marking it cancelled), and if it is already running tell sqlite to cancel it.
+    // At any point the operation could complete on another thread, so it is held weakly.
+    public func cancel(_ weakDBOperationGetter: () -> Cancellable?) {
+        weak var databaseOperation = weakDBOperationGetter()
```

Comment:
I think it's a good idea to, at the very top of this function, call `suspend` on the DB queue. That will prevent new tasks from executing.

The thing I'm afraid of is that we will enter `cancel` on a multi-core device, and end up interrupting the wrong operation.

The suspend in conjunction with the `running` check means that a `cancel` will either affect this `databaseOperation` _or nothing_, rather than this one _or the next one_.

Naturally we must call `queue.resume()` after the cancel occurs.

Reply:
Agreed, same concerns here.